### ISSUE TITLE
Having to walk all the subscribers on subscribe isn't great

### ISSUE THREAD

**Issue by Stebalien** [Kubuxu/go-broadcast#1]
https://github.com/Kubuxu/go-broadcast/blob/3ff470a934673de8a1eb405cde536f7f591e2244/broadcast.go#L20-L24

Probably fine for us... but this is a reason to consider creating/returning the channel internally.

**Comment by Kubuxu** [comment]
I agree it is not great, but it should not be an issue with less than 1k subscribers, and for at least our use-case, registration is not the hot path.
